drivers/hmis/app/graphql/types/hmis_schema/ce_referral_step.rb

```diff
@@ -8,37 +8,79 @@
 
 module Types
   class HmisSchema::CeReferralStep < Types::BaseObject
-    # object is a Hmis::WorkflowExecution::Step
+    # object is an OpenStruct with:
+    # {
+    #   step: Hmis::WorkflowExecution::Step,
+    #   referral: Hmis::Ce::Referral,
+    # }
+    # Why: The Step schema object should resolve this step's potential Participants (ReferralParticipants on the Referral).
+    # However, the Step model does not have a direct relationship to Referral, since it lives under Workflow Execution and not CE.
+    # Alternatives considered:
+    # todo @martha - reduce/remove this comment after we discuss and resolve.
+    # - Add relationship from Step to Referral, probably through Instance.
+    #   Decided against this in order to avoid a duplicative bidirectional relationship.
+    # - Always resolve the Referral alongside the Step, and expect the frontend to use the Participants from the Referral.
+    #   Decided against this because we want the frontend schema shape to match what would be most useful for the UI to display the data,
+    #   so we do want to put the participant on the Step.
 
     field :id, ID, null: false, description: 'unique identifier for this step based on node and instance'
     field :step_id, ID, null: true, method: :id, description: 'the DB identifier of this step, if it is persisted'
     field :form_definition, Types::Forms::FormDefinition, null: false
     field :name, String, null: false
     field :status, HmisSchema::Enums::CeReferralStepStatus, null: false
-    field :swimlane, String, null: false
     field :submitted_values, JsonObject, null: true
-    delegate :name, to: :workflow_node
+    field :swimlane, HmisSchema::CeReferralSwimlane, null: true, description: 'Swimlane for this step, which holds information about potential step participants'
```

Comment:
This is a breaking change (swimlane: `String` -> `HmisSchema::CeReferralSwimlane`)

Reply:
Thanks. I'm wondering about the nullability. In what situation does a step not have a swimlane? Is that something we actually want to support in the CE referral workflow? (I didn't think so, but maybe I'm missing something).

If we do want all tasks to have a swimlane, it's something we should enforce when allowing people to configure their own workflow templates. For now, since we're setting them up ourselves, maybe we can just raise  an exception if there is no swimlane associated with this step's task node.